### PR TITLE
Fix MONGOID-5033 Cannot use any_of with multiple conditions when using symbol operators on Time fields and passing Date instances


### PR DESCRIPTION
**Pull request by p-mongo**

### CODE REVIEW

**Comment by p-mongo** [comment]
https://jira.mongodb.org/browse/MONGOID-5033

**Review by durran — APPROVED** [review]
LGTM

File: docs/tutorials/mongoid-queries.txt [hunk 1]
```diff
@@ -1478,6 +1478,63 @@ It is also possible to query using PCRE syntax by constructing
...
+    
+    Voter.where(deregistered_at: Date.today).selector
+    # => {"deregistered_at"=>2020-12-18 00:00:00 UTC}
+
+When using the ``registered_at`` field which is of type ``Time``, the date
+was interpreted to be in local time (as per the :ref:`configured time zone
+<time-zones>`). When using the ``born_on`` field which is of type ``Date``,
+the date was interpreted to be in UTC. When using the ``voted_at`` field
+which was defined without a type (hence implicitly as an ``Object``),
+the date was used unmodified in the constructed query When using a
```
Comment (alcaeus) [inline docs/tutorials/mongoid-queries.txt:1533]:
```suggestion
the date was used unmodified in the constructed query. When using a
```

File: docs/tutorials/mongoid-queries.txt [hunk 2]
```diff
@@ -1478,6 +1478,63 @@ It is also possible to query using PCRE syntax by constructing
...
+nonexistent field ``deregistered_at`` the date was interpreted to be in UTC
+and converted to a time matching the behavior of querying a ``Date`` field.
```
Comment (alcaeus) [inline docs/tutorials/mongoid-queries.txt:1535]:
```suggestion
and converted to a time, matching the behavior of querying a ``Date`` field.
```

**Review by alcaeus — APPROVED** [review]

**Pull request merged**